Monika-After-Story/MonikaModDev
# Player Nicknames

WIP of player nickname system
# NOTE:
I'm still reviewing other script files to change more things to a nickname ref instead

## Review:
- Please verify nickname placements (ctrl+f `mas_get_player_nickname`) are okay
- Please indicate locations where some should be added (if any)
- Please verify the dialogue/exps are good for the nickname pick/edit topics

## Testing:
- Verify that `mas_get_player_nickname` will always return the `_default` value if affection is below `affectionate`
- Verify that `mas_get_player_nickname` will always return the `_default` value if the player hasn't set nicknames/allow none
- Verify that `mas_get_player_nickname` will never return a nickname in the `exclude_names` list (and the `_default` value if all names in the nickname pool are excluded)
- Verify that selecting valid nicknames works and the custom name entry also works (filtering according to the same rules as player name)
- Verify that editing valid nicknames works as well, bringing up the check scrollable containing all nicknames which were previously okay and then providing the ability to add more if you wish
- Verify no crashes
